tardigrade/roots/aws/baseline/main.tf

```diff
@@ -418,6 +393,182 @@ module "inspector" {
   tags             = local.tags
 }
 
+##### METRIC FILTERS #####
+locals {
+  # due to limitations in prowlers checking of filter_patterns within metric filters, very long strings
```

Comment:
Prowler checks the actual AWS resource config, not tf, right? So however it is represented here is immaterial, as long as it results in the same config in AWS that prowler is checking for. A heredoc is just a way to visually represent and manipulate a multi-line string. Perhaps the newlines were throwing things off. I'm not sure if you can use a heredoc inside a function directly (without using a local), but could try some variation of this:

```
      filter_pattern = replace(
        <<-EOL
          { ($.errorCode = "*UnauthorizedOperation") || ($.errorCode = "AccessDenied*") }
          EOL,
        "\n",
        ""
      )
```

Reply:
Yes, prowler checks the actual resource and yes the newlines is what's throwing things off. I previously attempted to do a replace with a heredoc but wasn't able to get it to work. Ended up trying to put all of the heredocs into a map but it was only recognizing the first element in the map. I can try again though.